src/compiler/program.ts

```diff
@@ -1376,7 +1376,7 @@ namespace ts {
             }
         }
 
-        function processSourceFile(fileName: string, isDefaultLib: boolean, refFile?: SourceFile, refPos?: number, refEnd?: number) {
+        function processSourceFile(fileName: string, isDefaultLib: boolean, isReference: boolean, refFile?: SourceFile, refPos?: number, refEnd?: number) {

```

Comment:
A little bit of documentation on this would be helpful. From a cursory glance, it doesn't seem obvious that this is differentiated from an import.

Alternatively, make an enum.